checks/packaging.go

```diff
@@ -77,49 +77,75 @@ func Packaging(c *checker.CheckRequest) checker.CheckResult {
 		"no packaging workflow used")
 }
 
-func isPackagingWorkflow(s, fp string, c *checker.CheckRequest) bool {
-	// nodejs packages
-	if strings.Contains(s, "uses: actions/setup-node@") {
+// A packaging workflow.
+func isPackagingWorkflow(s, fp string, dl checker.DetailLogger) bool {
+	// Nodejs packages.
+	if strings.Contains(s, "actions/setup-node@") {
 		r1 := regexp.MustCompile(`(?s)registry-url.*https://registry\.npmjs\.org`)
 		r2 := regexp.MustCompile(`(?s)npm.*publish`)
 
 		if r1.MatchString(s) && r2.MatchString(s) {
-			c.Dlogger.Info("candidate node packaging workflow using npm: %s", fp)
+			dl.Info("candidate node packaging workflow using npm: %s", fp)
 			return true
 		}
 	}
 
-	if strings.Contains(s, "uses: actions/setup-java@") {
+	// Java packages.
+	if strings.Contains(s, "actions/setup-java@") {
 		// Java packages with maven.
 		r1 := regexp.MustCompile(`(?s)mvn.*deploy`)
 		if r1.MatchString(s) {
-			c.Dlogger.Info("candidate java packaging workflow using maven: %s", fp)
+			dl.Info("candidate java packaging workflow using maven: %s", fp)
 			return true
 		}
 
 		// Java packages with gradle.
 		r2 := regexp.MustCompile(`(?s)gradle.*publish`)
 		if r2.MatchString(s) {
-			c.Dlogger.Info("candidate java packaging workflow using gradle: %s", fp)
+			dl.Info("candidate java packaging workflow using gradle: %s", fp)
 			return true
 		}
 	}
 
-	if strings.Contains(s, "actions/setup-python@") && strings.Contains(s, "pypa/gh-action-pypi-publish@master") {
-		c.Dlogger.Info("candidate python packaging workflow using pypi: %s", fp)
+	// Ruby packages.
+	r := regexp.MustCompile(`(?s)gem.*push`)
+	if r.MatchString(s) {
+		dl.Info("ruby publishing workflow using gem: %s", fp)
+		return true
+	}
+
+	// NuGet packages.
+	r = regexp.MustCompile(`(?s)nuget.*push`)
+	if r.MatchString(s) {
+		dl.Info("nuget publishing workflow: %s", fp)
 		return true
 	}
 
-	if strings.Contains(s, "uses: docker/build-push-action@") {
-		c.Dlogger.Info("candidate docker publishing workflow: %s", fp)
+	// Docker packages.
+	if strings.Contains(s, "docker/build-push-action@") {
+		dl.Info("candidate docker publishing workflow: %s", fp)
+		return true
+	}
+
+	r = regexp.MustCompile(`(?s)docker.*push`)
+	if r.MatchString(s) {
+		dl.Info("candidate docker publishing workflow: %s", fp)
+		return true
+	}
+
+	// Python packages.
+	if strings.Contains(s, "actions/setup-python@") && strings.Contains(s, "pypa/gh-action-pypi-publish@master") {
+		dl.Info("candidate python packaging workflow using pypi: %s", fp)
 		return true
 	}
 
-	if strings.Contains(s, "docker push") {
-		c.Dlogger.Info("candidate docker publishing workflow: %s", fp)
+	// Go packages.
```

Comment:
Let's add support for Rust/Cargo as well?

Reply:
done.